src/lib/resolvers/message.ts

```diff
@@ -34,58 +36,107 @@ export interface MessageResolverOptions {
 	scan?: boolean;
 }
 
-export async function resolveMessage(parameter: string, options: MessageResolverOptions): Promise<Result<Message, Identifiers.ArgumentMessageError>> {
-	const message =
-		(await resolveById(parameter, options)) ??
-		(await resolveByLink(parameter, options.message)) ??
-		(await resolveByChannelAndMessage(parameter, options.message));
-	if (message) return Result.ok(message);
+export async function resolveMessage(
+	parameter: IdResolvable,
+	options: MessageResolverOptions
+): Promise<Result<Message, Identifiers.ArgumentMessageError>> {
+	if (typeof parameter === 'string') {
+		const message =
+			(await resolveById(parameter, options)) ??
+			(await resolveByLink(parameter, options)) ??
+			(await resolveByChannelAndMessage(parameter, options));
+
+		if (message) {
+			return Result.ok(message);
+		}
+	} else {
+		const message = parameter.options.resolved.messages?.first();
```

Comment:
Can we add a comment here for the fact this comes only from context menu commands at this time?